src/w_file.c

```diff
@@ -0,0 +1,247 @@
+//
+// Copyright(C) 2024 Roman Fomin
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+#include <fcntl.h>
+#include <errno.h>
+
+#include "doomtype.h"
+#include "i_glob.h"
+#include "i_printf.h"
+#include "i_system.h"
+#include "m_io.h"
+#include "m_array.h"
+#include "m_misc.h"
+#include "m_swap.h"
+#include "w_internal.h"
+#include "w_wad.h"
+
+static int FileLength(int descriptor)
+{
+   struct stat st;
+
+   if (fstat(descriptor, &st) == -1)
+   {
+      I_Error("FileLength: failure in fstat\n");
+   }
+
+   return st.st_size;
+}
+
+static void W_FILE_AddDir(w_handle_t handle, const char *path,
+                          const char *start_marker, const char *end_marker)
+{
+    int startlump = numlumps;
+
+    glob_t *glob;
+
+    if (!strcmp(path, "."))
+    {
+        glob = I_StartGlob(handle.p1.base_path, "*.*",
+                           GLOB_FLAG_NOCASE | GLOB_FLAG_SORTED);
+    }
+    else
+    {
+        char *s = M_StringJoin(handle.p1.base_path, DIR_SEPARATOR_S, path, NULL);
+        glob = I_StartGlob(s, "*.*", GLOB_FLAG_NOCASE | GLOB_FLAG_SORTED);
+        free(s);
+    }
+
+    if (!glob)
+    {
+        return;
+    }
+
+    while (true)
```

Comment:
How about `I_Printf(VB_INFO, "Adding files from directory %s", path);`?

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1694/commits/db9375a6084b7ebcdade27bb2e9c6072b8b4697b